Allow the programs that are to be built to be defined in the build
configuration.

Provide an install target for installing the programs in $(INSTALL_DIR)/bin.

This patch by Thierry Reding.
Not submitted to Mesa.
--
Also remove references to GLU library since it is not required
and we don't actually build it at this point.


Index: mesa/progs/xdemos/Makefile
===================================================================
--- mesa.orig/progs/xdemos/Makefile	2007-11-11 00:34:12.000000000 +0100
+++ mesa/progs/xdemos/Makefile	2007-11-11 02:07:52.000000000 +0100
@@ -6,11 +6,11 @@
 
 INCDIR = $(TOP)/include
 
-LIB_DEP = $(TOP)/$(LIB_DIR)/$(GL_LIB_NAME) $(TOP)/$(LIB_DIR)/$(GLU_LIB_NAME)
+LIB_DEP = $(TOP)/$(LIB_DIR)/$(GL_LIB_NAME)
 
-LIBS = -L$(TOP)/$(LIB_DIR) -l$(GLU_LIB) -l$(GL_LIB) $(APP_LIB_DEPS)
+LIBS = -L$(TOP)/$(LIB_DIR) -l$(GL_LIB) $(APP_LIB_DEPS)
 
-PROGS = glthreads \
+PROGS ?= glthreads \
 	glxdemo \
 	glxgears \
 	glxgears_fbconfig \
@@ -46,6 +46,9 @@
 
 default: $(PROGS)
 
+install: $(PROGS)
+	$(INSTALL) -d $(INSTALL_DIR)/bin
+	$(INSTALL) -m 755 $(PROGS) $(INSTALL_DIR)/bin
 
 clean:
 	-rm -f $(PROGS)
Index: mesa/progs/Makefile
===================================================================
--- mesa.orig/progs/Makefile	2007-11-11 00:34:12.000000000 +0100
+++ mesa/progs/Makefile	2007-11-11 02:06:48.000000000 +0100
@@ -21,8 +21,12 @@
 		fi \
 	done
 
-# Dummy install target
 install:
+	@for dir in $(SUBDIRS) ; do \
+		if [ -d $$dir ] ; then \
+			(cd $$dir ; $(MAKE) install) ; \
+		fi \
+	done
 
 clean:
 	@for dir in $(SUBDIRS) tests ; do \
